runtime/drivers/druid/druidsqldriver/druid_api_sql_driver.go

```diff
@@ -40,193 +56,223 @@ type sqlConnection struct {
-			switch c {
-			case "TINYINT":
-				transformers[i] = func(v any) (any, error) {
-					switch v := v.(type) {
-					case float64:
-						return int8(v), nil
-					default:
-						return v, nil
-					}
-				}
-			case "SMALLINT":
-				transformers[i] = func(v any) (any, error) {
-					switch v := v.(type) {
-					case float64:
-						return int16(v), nil
-					default:
-						return v, nil
-					}
-				}
-			case "INTEGER":
-				transformers[i] = func(v any) (any, error) {
-					switch v := v.(type) {
-					case float64:
-						return int32(v), nil
-					default:
-						return v, nil
-					}
-				}
-			case "BIGINT":
-				transformers[i] = func(v any) (any, error) {
-					switch v := v.(type) {
-					case float64:
-						return int64(v), nil
-					default:
-						return v, nil
-					}
-				}
-			case "FLOAT":
-				transformers[i] = func(v any) (any, error) {
-					switch v := v.(type) {
-					case float64:
-						return float32(v), nil
-					case string:
-						return strconv.ParseFloat(v, 32)
-					default:
-						return v, nil
-					}
-				}
-			case "DOUBLE":
-				transformers[i] = func(v any) (any, error) {
-					switch v := v.(type) {
-					case string:
-						return strconv.ParseFloat(v, 64)
-					default:
-						return v, nil
-					}
-				}
-			case "REAL":
-				transformers[i] = func(v any) (any, error) {
-					switch v := v.(type) {
-					case string:
-						return strconv.ParseFloat(v, 64)
-					default:
-						return v, nil
-					}
+		req.Header.Add("Content-Type", "application/json")
+		resp, err := c.client.Do(req)
+		if err != nil {
+			var v *url.Error
+			if errors.As(err, &v) {
+				if errTooManyRedirects.MatchString(v.Error()) {
+					return nil, retrier.Fail, v
 				}
-			case "DECIMAL":
-				transformers[i] = func(v any) (any, error) {
-					switch v := v.(type) {
-					case string:
-						return strconv.ParseFloat(v, 64)
-					default:
-						return v, nil
-					}
+
+				if errInvalidProtocol.MatchString(v.Error()) {
+					return nil, retrier.Fail, v
 				}
-			case "TIMESTAMP":
-				transformers[i] = func(v any) (any, error) {
-					switch v := v.(type) {
-					case string:
-						t, err := time.Parse(time.RFC3339, v)
-						if err != nil {
-							return nil, err
-						}
-						return t, nil
-					default:
-						return v, nil
-					}
+
+				if errTLSCert.MatchString(v.Error()) {
+					return nil, retrier.Fail, v
 				}
-			case "ARRAY":
-				transformers[i] = func(v any) (any, error) {
-					var l []any
-					err := json.Unmarshal([]byte(v.(string)), &l)
-					if err != nil {
-						return nil, err
-					}
-					return l, nil
+
+				var x509err *x509.UnknownAuthorityError
+				if errors.As(v.Err, x509err) {
+					return nil, retrier.Fail, v
 				}
-			case "OTHER":
-				transformers[i] = func(v any) (any, error) {
-					var l map[string]any
-					err := json.Unmarshal([]byte(v.(string)), &l)
-					if err != nil {
-						return nil, err
-					}
-					return l, nil
+			}
+
+			return nil, retrier.Retry, err
```

Comment:
Does it make sense to retry when we don't even get a response back? For example, might just be the DSN that's broken or a local firewall, it's not great to have 126 seconds of retrying just to discover that.

Should we maybe opt for just retrying on the retryable HTTP status codes?

Reply:
Makes sense.